### PR TITLE
Remove unused code

### LINKED ISSUE #92: Was the unused code left by design, or was it perhaps an oversight?
I've noticed that there are some codes or variables that haven't been used. Is this by design, or might it have been an oversight?

The `objectLogDevice` ：

https://github.com/microsoft/garnet/blob/9ca34a25dfd71eedcc4e955254694fc17941cc3e/libs/storage/Tsavorite/cs/src/core/Allocator/SpanByteAllocator.cs#L414

The `newCurrentAddress` ：

https://github.com/microsoft/garnet/blob/9ca34a25dfd71eedcc4e955254694fc17941cc3e/libs/storage/Tsavorite/cs/src/core/TsavoriteLog/TsavoriteLogIterator.cs#L497

### PR DESCRIPTION
**Pull request by InCerryGit**
fix #92

### CODE REVIEW

**Comment by InCerryGit** [comment]
@microsoft-github-policy-service agree

**Review by Sooraz2 — APPROVED** [review]

File: libs/storage/Tsavorite/cs/src/core/Allocator/AllocatorBase.cs [hunk 1]
```diff
@@ -1909,7 +1907,7 @@ public void AsyncFlushPages(long fromAddress, long untilAddress, bool noFlush =
             long endPage = untilAddress >> LogPageSizeBits;
             int numPages = (int)(endPage - startPage);
 
-            long offsetInStartPage = GetOffsetInPage(fromAddress);
+            _ = GetOffsetInPage(fromAddress);
```
Comment (TedHartMS) [inline libs/storage/Tsavorite/cs/src/core/Allocator/AllocatorBase.cs:1910]: this line can just be removed

File: libs/storage/Tsavorite/cs/src/core/Allocator/AllocatorBase.cs [hunk 2]
```diff
@@ -2000,7 +1998,7 @@ public void AsyncFlushPages<TContext>(long flushPageStart, int numPages, DeviceI
         {
             for (long flushPage = flushPageStart; flushPage < (flushPageStart + numPages); flushPage++)
             {
-                int pageIndex = GetPageIndexForPage(flushPage);
+                _ = GetPageIndexForPage(flushPage);
```
Comment (TedHartMS) [inline libs/storage/Tsavorite/cs/src/core/Allocator/AllocatorBase.cs:2001]: This line can also be removed

**Review by TedHartMS — APPROVED** [review]
Nice cleanup, thanks!

**Pull request merged**